tests/solidity/test_fees.py

```diff
@@ -0,0 +1,367 @@
+    test.assertAlmostEqual(gov_balance_after, gov_fees_after_close_long, delta=1e-16 * test.trade_amount)
+
+
+@pytest.mark.parametrize("amount", AMOUNT, ids=idfn)
+def test_collect_fees_short(amount):
+    """Open a short and then close close to maturity; verify that gov fees are correct"""
+    test = TestFees(trade_amount=amount)  # set up test object
+
+    # check that both gov fees and gov balance are 0 before opening a short
+    gov_fees_before_open_short = get_gov_fees_accrued(test)
+    test.assertEqual(gov_fees_before_open_short, 0)
+    gov_balance_before_open_short = test.gary.wallet.balance.amount
+    test.assertEqual(gov_balance_before_open_short, 0)
+
+    # open short
+    test.hyperdrive.open_short(test.bob.wallet, test.trade_amount)
+    gov_fees_after_open_short = get_gov_fees_accrued(test)
+    test.assertGreater(gov_fees_after_open_short, gov_fees_before_open_short)
+
+    # hyperdrive into the future
+    warp(test, time_delta=0.5)
+
+    # close short
+    test.hyperdrive.close_short(
+        agent_wallet=test.bob.wallet,
+        bond_amount=test.bob.wallet.shorts[0].balance,
+        mint_time=0,
+        open_share_price=1,
+    )
+
+    # ensure that gov fees have increased
+    gov_fees_after_close_short = get_gov_fees_accrued(test)
+    test.assertGreater(gov_fees_after_close_short, gov_fees_after_open_short)
+
+    # collect fees to Governance Gary
+    test.hyperdrive.collect_gov_fee(test.gary.wallet)
+    gov_fees_after_collection = get_gov_fees_accrued(test)
+    test.assertEqual(gov_fees_after_collection, 0)
+
+    gov_balance_after = test.gary.wallet.balance.amount
+    # ensure that Governance Gary's balance has increased
+    test.assertGreater(gov_balance_after, gov_balance_before_open_short)
+    # ensure that Governance Gary got the exaxt fees expected
+    test.assertAlmostEqual(gov_balance_after, gov_fees_after_close_short, delta=1e-16 * test.trade_amount)
+
+
+def get_all_the_fees(test: TestFees, func: str) -> Tuple[float, float, float, float]:
+    """Get all the fees from the market state"""
+    # calculate time remaining
+    years_remaining = time.get_years_remaining(
+        market_time=test.hyperdrive.block_time.time,
+        mint_time=0,
+        position_duration_years=test.hyperdrive.position_duration.days / 365,
+    )  # all args in units of years
+    time_remaining = time.StretchedTime(
+        days=years_remaining * 365,  # converting years to days
+        time_stretch=test.hyperdrive.position_duration.time_stretch,
+        normalizing_constant=test.hyperdrive.position_duration.normalizing_constant,
+    ).normalized_time
+
+    # calculate curve fee
+    spot_price_adjusted = test.hyperdrive.spot_price
+    if func == "_calculateFeesOutGivenSharesIn":  # input is in shares
+        spot_price_adjusted = 1 / test.hyperdrive.spot_price  # when given shares, invert price
+
+    curve_fee, gov_curve_fee = test.hyperdrive.pricing_model.calc_curve_fee_split(
```

Comment:
should be able to get gov fee from trade breakdown in `calc_*_given_*`

Reply:
done in 00338cdf842d002780943bd00928077a6622826e